yggdrasil_decision_forests/learner/decision_tree/oblique.cc

```diff
@@ -788,12 +788,31 @@ void SampleProjection(const absl::Span<const int>& features,
     }
   };
 
+  #ifndef NDEBUG  // Keep DCHECK_EQ from for feature : features
   for (const auto feature : features) {
     DCHECK_EQ(data_spec.columns(feature).type(), dataset::proto::NUMERICAL);
-    if (unif01(*random) < projection_density) {
-      projection->push_back({feature, gen_weight(feature)});
-    }
   }
+  #endif
+
+  const size_t p = features.size();
+  std::binomial_distribution<size_t> binom(p, projection_density);
+
+  // Exp[Binomial(p,projection_density)] = k
+  const size_t k = binom(*random);
+
+  absl::btree_set<size_t> picked_idx;
+
+  // Floyd's sampler to select k indices uniformly
+  for (size_t j = p - k; j < p; ++j) {
+    size_t t = absl::Uniform<size_t>(*random, 0, j + 1);
```

Comment:
Feature/column indexes are encoded as int32.
I wonder if generating a random int32 (32bits) would be faster than generating a random size_t (i.e. uint64 in most usecase).

Reply:
*The YDF team will handle it after this PR is submited